qiskit_nature/circuit/library/initial_states/fermionic_gaussian_states.py

```
@@ -0,0 +1,361 @@
+
+    def __init__(
+        self,
+        transformation_matrix: np.ndarray,
+        qubit_converter: Optional[QubitConverter] = None,
+        **circuit_kwargs,
+    ) -> None:
+        r"""Initialize a circuit that prepares a Slater determinant.
+
+        A Slater determinant is a state of the form
+
+        .. math::
+            b^\dagger_1 \cdots b^\dagger_{N_f} \lvert \text{vac} \rangle
+
+        where
+
+        .. math::
+            b^\dagger_j = \sum_{k = 1}^N Q_{jk} a^\dagger_k,
+
+        - :math:`Q` is an :math:`N_f \times N` matrix with orthonormal rows
+        - :math:`a^\dagger_1, \ldots, a^\dagger_{N}` are the fermionic creation operators
+        - :math:`\lvert \text{vac} \rangle` is the vacuum state
+          (mutual 0-eigenvector of the fermionic number operators :math:`\{a^\dagger_j a_j\}`)
+
+        Currently, only the Jordan-Wigner Transformation is supported.
+
+        Reference: arXiv:1711.05395
+
+        Args:
+            transformation_matrix: The matrix :math:`Q` that specifies the coefficients of the
+                new creation operators in terms of the original creation operators.
+                The rows of the matrix must be orthonormal.
+            qubit_converter: A QubitConverter instance.
+            circuit_kwargs: Keyword arguments to pass to the QuantumCircuit initializer.
+
+        Raises:
+            ValueError: transformation_matrix must be a 2-dimensional array.
+            ValueError: transformation_matrix must have orthonormal rows.
+            NotImplementedError: Currently, only the Jordan-Wigner Transform is supported.
+        """
+        if not len(transformation_matrix.shape) == 2:
+            raise ValueError("transformation_matrix must be a 2-dimensional array.")
+
+        m, n = transformation_matrix.shape
+        if m > n:
+            raise ValueError("transformation_matrix must have orthonormal rows.")
+        # TODO maybe actually check if the rows are orthonormal
```

I would expect this if you want to throw the above error.

---

I changed the error message.